### PR TITLE
u-boot: configure boot partition based on configuration

### PR DESCRIPTION
**Pull request by grandpaul**
Add a variable UBOOT_DEFAULT_BOOT_PARTITION to control which partition
looks for boot.scr

IOTMBL-2416: BSP Team: Make partition on which U-Boot looks for FIT image
configurable at build/config time

Signed-off-by: Ying-Chun Liu (PaulLiu) <paulliu@debian.org>

### CODE REVIEW

**Comment by grandpaul** [comment]
Manual test: 
rpi3 https://paste.ubuntu.com/p/FMwPV4mtPy/
warp7 https://paste.ubuntu.com/p/rhVTv5nbbB/
imx8 https://paste.ubuntu.com/p/k83tCkxq4K/
pico7 https://paste.ubuntu.com/p/rhVTv5nbbB/

**Comment by jonathanhaigh-arm** [comment]
Can you post a Jenkins link please?

**Comment by grandpaul** [comment]
Old Jenkins.
http://jenkins.mbed-linux.arm.com/view/paulliu/job/paulliu-00-jenkins-test-1/22/

Let me fix the UBOOT_DEFAULT_BOOT_PARTITION ?= "1" and rebuild again.

File: openembedded-core-mbl/meta/classes/mbl-uboot-sign.bbclass [hunk 1]
```diff
@@ -12,6 +12,9 @@
 UBOOT_DTB_IMAGE ?= "u-boot-${MACHINE}-${PV}-${PR}.dtb"
 UBOOT_NODTB_IMAGE ?= "u-boot-nodtb-${MACHINE}-${PV}-${PR}.${UBOOT_SUFFIX}"
 
+# Default partitions to read the boot.scr (FIT image)
+UBOOT_DEFAULT_BOOT_PARTITION ?= "1"
```
Comment (jonathanhaigh-arm) [inline openembedded-core-mbl/meta/classes/mbl-uboot-sign.bbclass]: I'm not keen on this being added to a class called `mbl-uboot-sign` - it doesn't have anything to do with signing.

If you want to put this somewhere that applies to all of the U-Boot recipes then you could add it to `meta-mbl-bsp-common/recipes-bsp/u-boot/u-boot%.bbappend`

Comment (grandpaul) [reply openembedded-core-mbl/meta/classes/mbl-uboot-sign.bbclass]: ok. I'll put it there.

**Comment by grandpaul** [comment]
jenkins:
http://jenkins.mbed-linux.arm.com/view/paulliu/job/paulliu-00-jenkins-test-1/24/

**Pull request merged**